### ISSUE TITLE
why set require gradient False

### ISSUE THREAD

**Issue by vanpersie32** [ruotianluo/self-critical.pytorch#35]
Hi, much thanks for your code. I have question that why do you set requires_grad to False in [here](https://github.com/ruotianluo/self-critical.pytorch/blob/master/models/AttModel.py#L185-L188)

**Comment by ruotianluo** [comment]
I wrote this when I wasn't sure about the default value.

**Comment by vanpersie32** [comment]
Yeah, I also have question here. Do we need to backward gradients to the model through the "it" in [here](https://github.com/ruotianluo/self-critical.pytorch/blob/master/models/AttModel.py#L185-L188). What I mean is that the loss backward gradients to it, then it backward gradients to the model.

**Comment by ruotianluo** [comment]
no, it doesn't

**Comment by vanpersie32** [comment]
Thanks for replying
